src/main.rs

```
@@ -13,424 +13,50 @@ extern crate serde; // JSON
-
-        // fill the vector with query's result
-        for row in stmt.query(&[&tl_long, &br_long, &tl_lat, &br_lat]).unwrap() {
-            pictures.push(PictureDBData {
-                id: row.get("id"),
-                author: row.get("author"),
-                description: row.get("description"), // optional
-                gps_lat: row.get("gps_lat"),
-                gps_long: row.get("gps_long"),
-                date_taken: format_date(&row.get("date_taken")),
-                rating: row.get("rating"), // optional
-                likes: row.get("likes"),
-            });
-        }
-
-        println!("[*] Instruction exectued");
-
-        serde_json::ser::to_string(&pictures).unwrap() // return the json value of pictures vec
-    }});
-
-
-    // Accepts only JSON
-    server.post("/pictures", middleware! { |req, mut res| {
-        /*
-            inserting picture's metadata into the database.
-            the API returns the id of the created row, and returns this id.
-            the client then needs to upload the picture.
-        */
-
-        res.set(MediaType::Json); // HTTP header : Content-Type: application/json (for return)
-
-        let conn = req.db_conn();
-        // retreive the metadata in JSON
-        let pic_metadata = req.json_as::<PictureMetadata>().unwrap();
-
-        let stmt = conn.prepare("INSERT INTO pictures
-                                (author, description, gps_lat, gps_long, date_taken, rating, uploaded)
-                                VALUES($1, $2, $3, $4, NOW(), $5, FALSE)
-                                RETURNING id").unwrap();
-        let query = stmt.query(&[&pic_metadata.author,
-                                &pic_metadata.description,
-                                &pic_metadata.gps_lat,
-                                &pic_metadata.gps_long,
-                                &pic_metadata.rating]);
-        let rows = query.iter()
```

This was basically the same as `query.ok().unwrap()`, just much less clearer. Both are worse than unwrapping directly because they hide the error type.